app/models/stamp.rb

```diff
@@ -24,7 +24,7 @@ def deactivate!
 
   # Name must be unique within the scope of active stamps
   def validate_name_active_uniqueness
-    if Stamp.where(name: name, deactivated_at: nil).any? { |s| s.id != id }
+    if Stamp.where("name = ? AND deactivated_at is NULL", name).exists?
```

Comment:
Wouldn't this always be true on update (if the name hasn't changed)?

Reply:
`where(name: name, deactivated_at: nil).any? { |s| s.id != id }` looks correct. The new version looks wrong. Why was it changed?

Reply:
I thought it might be more efficient to do it in a single query... something like `Stamp.where("name = ? AND id != ? AND deactivated_at is NULL", name, id).exists?` rather than loop through all stamps with that name. Tbh probably never going to be much in it.